File: app/src/main/java/co/smartreceipts/android/imports/AutoFailImportProcessor.java
package co.smartreceipts.android.imports;

import android.net.Uri;
import androidx.annotation.NonNull;

import java.io.File;

import io.reactivex.Single;

public class AutoFailImportProcessor implements FileImportProcessor {
    @NonNull
    @Override
    public Single<File> process(@NonNull Uri uri) {
        return Single.error(new UnsupportedOperationException());
    }
}

File: app/src/main/java/com/wops/receiptsgo/imports/AutoFailImportProcessor.kt
package com.wops.receiptsgo.imports

import android.net.Uri
import io.reactivex.Single
import java.io.File

class AutoFailImportProcessor : FileImportProcessor {

    override fun process(uri: Uri): Single<File> {
        return Single.error(UnsupportedOperationException())
    }
}
